### PR TITLE
[RW-4429][risk=no] Add age type options in CB age wizard

### PR DESCRIPTION
**Pull request by dolbeew**
- Adds radio buttons with three age type options to search by: current age, age at consent and age at CDR date.
- Loads counts for all ages and age types so we can calculate the current count as the user changes the age slider
- Both changes are behind the `enableCBAgeTypeOptions` feature flag (enabled only in local env)

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/ConfigController.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
             .unsafeAllowSelfBypass(config.access.unsafeAllowSelfBypass)
             .enableNewAccountCreation(config.featureFlags.enableNewAccountCreation)
             .requireInvitationKey(config.access.requireInvitationKey)
-            .requireInstitutionalVerification(
-                config.featureFlags.requireInstitutionalVerification));
+            .requireInstitutionalVerification(config.featureFlags.requireInstitutionalVerification)
+            .enableCBAgeTypeOptions(config.featureFlags.enableCBAgeTypeOptions));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/ConfigController.java]: Why do we need this?

File: api/src/main/resources/workbench.yaml [hunk 2]
```diff
@@ -2483,6 +2483,10 @@ definitions:
         description: >
           Feature flag for verifying the institutional affiliation of new users by validating their
           contact emails.
+      enableCBAgeTypeOptions:
```
Comment (freemabd) [inline api/src/main/resources/workbench.yaml]: Why do we need this?

File: common-api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java [hunk 3]
```diff
@@ -237,6 +237,8 @@ public String freeTierBillingAccountName() {
     // enforced by pattern-matching the user's contact email against the institution's
     // set of whitelisted email domains or addresses
     public boolean requireInstitutionalVerification;
+    // Flag to indicate whether to use the new age type options in cohort builder age wizard
+    public boolean enableCBAgeTypeOptions;
```
Comment (freemabd) [inline common-api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java]: Why do we need this?

File: api/src/main/java/org/pmiops/workbench/api/ConfigController.java (hunk 1 above)
Comment (dolbeew) [reply api/src/main/java/org/pmiops/workbench/api/ConfigController.java]: Per [this comment](https://github.com/all-of-us/workbench/blob/master/ui/src/environments/environment-type.ts#L59), that is now the preferred way to add feature flags.

File: api/src/main/resources/workbench.yaml (hunk 2 above)
Comment (dolbeew) [reply api/src/main/resources/workbench.yaml]: See above response

File: common-api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java (hunk 3 above)
Comment (dolbeew) [reply common-api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java]: See above response

**Comment by gjuggler** [comment]
> Looks good. Few comments about server side flags. We don't need the server side flags. Seems like client side flags are enough.

I may be somewhat responsible for this – we're trying to unify our flag control to be only based on the server-side config. It became too confusing when some flags were in environment.ts, while others were in WorkbenchConfig.

It's also easier to unit test variations of flag configs when they're not hard-coded into environment.ts.

I added this loud warning in environment.ts: https://github.com/all-of-us/workbench/blob/master/ui/src/environments/environment-type.ts#L59

**Comment by freemabd** [comment]
> > Looks good. Few comments about server side flags. We don't need the server side flags. Seems like client side flags are enough.
> 
> I may be somewhat responsible for this – we're trying to unify our flag control to be only based on the server-side config. It became too confusing when some flags were in environment.ts, while others were in WorkbenchConfig.
> 
> It's also easier to unit test variations of flag configs when they're not hard-coded into environment.ts.
> 
> I added this loud warning in environment.ts: https://github.com/all-of-us/workbench/blob/master/ui/src/environments/environment-type.ts#L59

Thanks for the background Greg, this makes much more sense now. Seems like the right thing to do Will.

**Pull request merged**